lib/api.js

```diff
@@ -376,15 +374,23 @@ api.declare({
   // instance is not managed, and should not be authorized.  If there's more
   // than one, the database is not providing the relational guarantees it
   // should
-  if (!req.satisfies({region, instanceId})) { 
+  try {
+    await req.authorize({region, instanceId, hasWorkerType: false});
+  } catch (err) {
+    if (err.code !== 'AuthorizationError') {
+      throw err;
+    }
     let workerTypes = await this.state.listInstances({id: instanceId, region});
     switch (workerTypes.len != 1) {
```

Comment:
not realted to this PR, but this should be a switch on `workerTypes.length`.  This is not a security issues because `workerTypes.len` should be an undefined property as the return value of that function is a list.  When `workerTypes.len` (undefined) is `!=`'d to `1`, it's true.  Because the switch is a switch on `true`, only the default case matches.

I can take a look at this later, but I'd like to fix this before we land this patch

See also:
```
> undefined != 1
true
> [].len
undefined
> switch (true) { case 0: console.log('zero'); break; case 1: console.log('one'); break; default: console.log('default')}
default
```